codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/OperationRegistryGenerator.kt

```diff
@@ -0,0 +1,172 @@
+            "HttpBody" to RuntimeType.AxumHttpBody,
+            "async_trait" to RuntimeType.AsyncTrait,
+            "RequestParts" to RuntimeType.RequestParts
+        )
+    }
+
+    private fun OperationShape.fromRequestIntoResponseWorkaround(): String {
+        val input = symbolProvider.toSymbol(this.inputShape(model))
+        val output = symbolProvider.toSymbol(this.outputShape(model))
+
+        return """
+           impl #{IntoResponse} for $output {
+                type Body = #{Body};
+                type BodyError = <Self::Body as #{HttpBody}>::Error;
+
+                fn into_response(self) -> http::Response<Self::Body> {
+                    todo!();
+                }
+            }
+
+            ##[#{async_trait}]
+            impl<B> #{FromRequest}<B> for $input
+            where
+                B: Send, // required by `async_trait`
+            {
+                type Rejection = http::StatusCode;
+
+                async fn from_request(
+                    _req: &mut #{RequestParts}<B>,
+                ) -> Result<Self, Self::Rejection> {
+                    todo!()
+                }
+            }
+        """.trimIndent()
+    }
+
+    private fun OperationShape.requestSpec(): String {
+        val httpTrait = httpBindingResolver.httpTrait(this)
+        val namespace = "$serverCrate::routing::request_spec"
+
+        // TODO Support the `endpoint` trait: https://awslabs.github.io/smithy/1.0/spec/core/endpoint-traits.html#endpoint-trait
+
+        val pathSegments = httpTrait.uri.segments.map {
+            "$namespace::PathSegment::" +
+                if (it.isGreedyLabel) "Greedy"
+                else if (it.isLabel) "Label"
+                else "Literal(String::from(\"${it.content}\"))"
+        }
+        val querySegments = httpTrait.uri.queryLiterals.map {
+            "$namespace::QuerySegment::" +
+                if (it.value == "") "Key(String::from(\"${it.key}\"))"
+                else "KeyValue(String::from(\"${it.key}\"), String::from(\"${it.value}\"))"
+        }
+
+        // TODO Note I'm writing out the fully namespaced names of the types, which one would do reaching for `RuntimeType`.
+        //  I don't foresee that we will use these types anywhere else in the codebase, so is it really worth adding them
+        //  to `RuntimeType.kt`, or creating them here and converting them to strings using `fullyQualifiedName()`,
+        //  as opposed to just interpolating strings like I'm doing here? I get the value of using `RuntimeType` when it's
+        //  an ubiquitous type used throughout the codebase, since it means it's definition is centralized, making it easy
+        //  to modify, but if the use of a type is localized like here, is there any benefit? Interpolating strings reads
+        //  clearer in my opinion.
+        //  I guess there has to be at least one usage of `RuntimeType`; otherwise the dependency won't appear in `Cargo.toml`.
```

Comment:
1. yes, the main reason to use `RuntimeType` is so that you ensure you add the dependency, but only when it's actually used. As the project gets more complex, you may end up with certain services using differing subsets of dependencies
2. yes, you can call `addDependency` directly on the writer